### ISSUE TITLE
Editing description doesn't work anymore with 1.0.2 & Firefox

### ISSUE THREAD

**Issue by simonspa** [nextcloud/deck#1979]
**Describe the bug**
I just updated to Deck 1.0.2 and tried to edit the description of a card but the editor doesn't show up. I made sure this is not a browser cache issue.

This seems to only affect Firefox, on Chrome I still get the description editor.

**To Reproduce**
1. Update
2. Click on a card
3. Click on the "edit" icon
4. Observe missing editor

**Expected behavior**
Editor should appear

**Screenshots**

Viewing (existing description):

![image](https://user-images.githubusercontent.com/1677436/83640351-4c394100-a5ac-11ea-866a-0222d7ebef62.png)

Clicking on editing:

![image](https://user-images.githubusercontent.com/1677436/83640398-5c512080-a5ac-11ea-9f78-13b7df6e47be.png)

**Client details:**
 - OS: Ubuntu 20.04
 - Browser: Firefox
 - Version: 76
 - Device: Desktop

<details>
<summary>Server details</summary>

**Operating system**: Debian 10

**Web server:** nginx 1.14

**Database:** MariaDB 10.3

**PHP version:** 7.3.14

**Nextcloud version:** 18.0.4

**Where did you install Nextcloud from:** Manual

**Signing status:**

```
No errors have been found.
```

**List of activated apps:**

```
 sudo -u www-data php occ app:list
Enabled:
  - accessibility: 1.4.0
  - activity: 2.11.0
  - admin_audit: 1.8.0
  - apporder: 0.10.0
  - bookmarks: 3.1.0
  - bruteforcesettings: 1.6.0
  - calendar: 2.0.3
  - cloud_federation_api: 1.1.0
  - comments: 1.8.0
  - contacts: 3.3.0
  - cospend: 0.5.4
  - dav: 1.14.0
  - deck: 1.0.2
  - drawio: 0.9.5
  - event_update_notification: 1.0.2
  - federatedfilesharing: 1.8.0
  - federation: 1.8.0
  - files: 1.13.1
  - files_downloadactivity: 1.7.0
  - files_external: 1.9.0
  - files_markdown: 2.2.0
  - files_pdfviewer: 1.7.0
  - files_rightclick: 0.15.2
  - files_sharing: 1.10.1
  - files_trashbin: 1.8.0
  - files_versions: 1.11.0
  - files_videoplayer: 1.7.0
  - firstrunwizard: 2.7.0
  - impersonate: 1.5.1
  - logreader: 2.3.0
  - lookup_server_connector: 1.6.0
  - metadata: 0.11.1
  - news: 14.1.11
  - notes: 3.4.0
  - notifications: 2.6.0
  - oauth2: 1.6.0
  - password_policy: 1.8.0
  - passwords: 2020.5.0
  - photos: 1.0.0
  - previewgenerator: 2.3.0
  - privacy: 1.2.0
  - provisioning_api: 1.8.0
  - quota_warning: 1.7.0
  - serverinfo: 1.8.0
  - settings: 1.0.0
  - sharebymail: 1.8.0
  - spreed: 8.0.9
  - tasks: 0.13.1
  - text: 2.0.0
  - theming: 1.9.0
  - theming_customcss: 1.6.0
  - twofactor_admin: 2.1.0
  - twofactor_backupcodes: 1.7.0
  - twofactor_totp: 4.1.3
  - twofactor_u2f: 5.1.0
  - updatenotification: 1.8.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0
Disabled:
  - encryption
  - nextcloud_announcements
  - recommendations
  - support
  - survey_client
  - systemtags
  - user_ldap
```

**Nextcloud configuration:**

```
 sudo -u www-data php occ config:list system
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "data-fingerprint": "58897020af7835d5d5eebe9f70033c28",
        "dbtype": "mysql",
        "version": "18.0.4.2",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "3306",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "htaccess.RewriteBase": "\/",
        "mail_smtpsecure": "ssl",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "logtimezone": "Europe\/Berlin",
        "log_rotate_size": "104857600",
        "mail_smtpauthtype": "LOGIN",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "memcache.local": "\\OC\\Memcache\\APCu",
        "maintenance": false,
        "preview_max_y": 2048,
        "preview_max_x": 2048,
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "mail_smtpauth": 1,
        "simpleSignUpLink.shown": false,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "knowledgebaseenabled": false,
        "loglevel": 2,
        "filelocking.enabled": "true",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "updater.release.channel": "stable",
        "activity_expire_days": 90,
        "mail_smtpmode": "smtp",
        "theme": ""
    }
}
```

**Are you using an external user-backend, if yes which one:** Nope
</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
Nothing of relevance

#### Browser log
Nothing

</details>

**Comment by juliushaertl** [comment]
Does clearing the browser cache fix the issue?

**Comment by juliushaertl** [comment]
Aw sorry, to fast, but I cannot reproduce that. It works fine here in Firefox. Any additional steps you taken after loading the page, or is it just:
- Load the page
- Open a card
- Click the edit button

**Comment by simonspa** [comment]
It really is just that. Also, after clicking the edit button, also the "close" doesn't work anymore:

![Peek 2020-06-03 15-47](https://user-images.githubusercontent.com/1677436/83644501-a4267680-a5b1-11ea-83be-c0553b82b44f.gif)

**Comment by juliushaertl** [comment]
> also the "close" doesn't work anymore

Then it seems there is some error crashing entire component, is the browser console really not reporting any issue?

**Comment by simonspa** [comment]
That's the only things I see:

![image](https://user-images.githubusercontent.com/1677436/83645023-40e91400-a5b2-11ea-9ffd-077dc782bfd5.png)

**Comment by AaronADev** [comment]
Having the exact same issue but both in Firefox and in Chrome

**Comment by porjo** [comment]
Works for me:
Firefox 76.0.1
Deck 1.0.2
Nextcloud server 18.0.4
OS: Fedora 32

**Comment by bendilobi** [comment]
Same bug here on Google Chrome for Windows Version 83.0.4103.97. 

As soon as I click the Edit button, two errors appear in the Console:
`TypeError: Cannot read property 'mimetype' of undefined
    at AttachmentList.vue?bd85:1
    at i.bt [as _l] (vue.esm.js:2639)
    at i.<anonymous> (AttachmentList.vue?bd85:1)
    at i.e._render (vue.esm.js:3557)
    at i.r (vue.esm.js:4075)
    at mn.get (vue.esm.js:4488)
    at mn.run (vue.esm.js:4563)
    at pn (vue.esm.js:4319)
    at Array.<anonymous> (vue.esm.js:1989)
    at Ke (vue.esm.js:1915)`
`Error: marked(): input parameter is of type [object Object], string expected
    at new ce (marked.js:2146)
    at vue.esm.js:3292
    at init (vue.esm.js:3123)
    at vue.esm.js:5989
    at d (vue.esm.js:5936)
    at vue.esm.js:6227
    at M (vue.esm.js:6330)
    at i.__patch__ (vue.esm.js:6493)
    at i.e._update (vue.esm.js:3957)
    at i.r (vue.esm.js:4075)`

**Comment by f512** [comment]
Same here with Brave Browser. Refreshing cache does not change anything. But removing cookies does correct the problem.
When I click the Edit button, I don't have any error in the console. The #description-preview div is simply replaced by "".
When I click on the eye again, I go back to the description preview.

**Comment by simonspa** [comment]
Interestingly, here deleting cookies doesn't help (apart from me running for my second factor to log in again... :rofl: )

I still have the same problem.

**Comment by a1pilot** [comment]
Seeing the same issue in Windows 10, Chrome Version 83.0.4103.97 (Official Build) (64-bit).

The eye button doesn't work at all.

I've also noticed the close button doesn't work on the details window (cross in top right), the title of the card glitches, and I can't change to another card. Have to refresh the window to get functionality back. It's as if the whole app crashes.

**Comment by a1pilot** [comment]
Error in consle, "main.js:62

![image](https://user-images.githubusercontent.com/665111/83805762-5a6c8780-a6a8-11ea-8604-c26851a41dff.png)

**Comment by linucksrox** [comment]
I don't mean to add more noise to this thread, but just wanted to point out I'm having this exact issue with Firefox 68.9.0esr, Deck 1.0.2, and NC 18.0.4 (and now 18.0.5). However, same browser, same Deck version with my NC19 server editing works as expected.

TLDR; broken in NC 18.x but working in NC19

**Comment by marcodegroot** [comment]
I've had the same error (Error: marked(): input parameter is of type [object Object], string expected) on 18.0.4. After a logout/login in Nextcloud, the issue disappeared for me. Now I can edit the descriptions again without a problem.

**Comment by linucksrox** [comment]
@marcodegroot thanks, I can confirm that works for me (well, I also cleared all cookies before logging back in) but it's working now.

**Comment by olegbolden** [comment]
Clearing cookies doesn't help me. (Chrome, NC 18.0.5)
I described problems at NC's help forum:
https://help.nextcloud.com/t/1-0-2-doesnt-work-in-nc-18-0-5/83721

**Comment by rkaraba** [comment]
Logout and clearing cookies doesn't help me. (Windows 10, Chrome and Firefox, NC 18.0.4)

**Comment by jansppenrade2** [comment]
Same issue here.

Browser: Google Chrome 81.0.4044.122
Nextcloud server version: 18.0.4
Deck version: 1.0.2

**Comment by nanocat-net** [comment]
I just experienced this in Chrome, 83.0.4103.61 on Debian 10. I am adding my notes about it in case it helps:

When I click on a card to look at the description, and then click the pencil button to edit the description, the pencil turns into an eye like normal, but all of the description text disappears. When I click on the eye to stop editing, the description text is still gone, and the eye doesn't turn back into a pencil. ALSO, the title at the top of the card gets visually scrambled.

**Comment by amsaravi** [comment]
> I just experienced this in Chrome, 83.0.4103.61 on Debian 10. I am adding my notes about it in case it helps:
> 
> When I click on a card to look at the description, and then click the pencil button to edit the description, the pencil turns into an eye like normal, but all of the description text disappears. When I click on the eye to stop editing, the description text is still gone, and the eye doesn't turn back into a pencil. ALSO, the title at the top of the card gets visually scrambled.
> 
> ... the Description editor field works properly after I enable "Disable cache" at the top of the Network tab of the Developer Console (ctrl-shift-i)

i have same problem and disabling cash solved that. thx for your workaround.

**Comment by HieronymusCH** [comment]
> Example of glitched title after clicking the edit description button:
> ![image](https://user-images.githubusercontent.com/665111/83806038-c8b14a00-a6a8-11ea-9b06-8f8020fb57e8.png)
> EDIT: this just happens and isn't related to the edit description button. This is another separate issue.
> 
> No errors when trying to change to another card. It just doesn't work.

I can confirm the glitch in the card title.

NextCloud: 19.0.0
Deck: 1.0.2
Firefox: 77.0.1 (64-Bit)

**Comment by olegbolden** [comment]
> > Example of glitched title after clicking the edit description button:
> > ![image](https://user-images.githubusercontent.com/665111/83806038-c8b14a00-a6a8-11ea-9b06-8f8020fb57e8.png)
> > EDIT: this just happens and isn't related to the edit description button. This is another separate issue.
> > No errors when trying to change to another card. It just doesn't work.
> 
> I can confirm the glitch in the card title.
> 
> NextCloud: 19.0.0
> Deck: 1.0.2
> Firefox: 77.0.1 (64-Bit)

This is an CSS issue. It can be easely fixed if you remove `height:52px` statement from `/apps/deck/js/deck.js` file. I attached the fixed one for NC18 here.
[deck.js.zip](https://github.com/nextcloud/deck/files/4784535/deck.js.zip)
